src/adafruit_blinka/microcontroller/tegra/pin.py

```diff
@@ -0,0 +1,132 @@
+import sys
+import Jetson.GPIO as GPIO
+sys.path.append("/opt/nvidia/jetson-gpio/lib/python")
+sys.path.append("/opt/nvidia/jetson-gpio/lib/python/Jetson/GPIO")
+GPIO.setmode(GPIO.BCM)
+GPIO.setwarnings(False)   # shh!
+
+# Each Jetson model uses different I2C busses
+JETSON_I2C_BUS_DEFS = {
+    "JETSON_TX1": [0, 1],
+    "JETSON_TX2": [1, 0],
+    "JETSON_XAVIER": [8, 1],
+    "JETSON_TXX": [1, 0]
+}
+
+model = GPIO.get_model()
+I2C_BUS = JETSON_I2C_BUS_DEFS[model][0]
+I2C_BUS_1 = JETSON_I2C_BUS_DEFS[model][1]
+
+# Pins dont exist in CPython so...lets make our own!
+class Pin:
+    IN = 0
+    OUT = 1
+    LOW = 0
+    HIGH = 1
+    PULL_NONE = 0
+    PULL_UP = 1
+    PULL_DOWN = 2
+
+    id = None
+    _value = LOW
+    _mode = IN
+
+    def __init__(self, bcm_number):
+        self.id = bcm_number
+
+    def __repr__(self):
+        return str(self.id)
+
+    def __eq__(self, other):
+        return self.id == other
+
+    def init(self, mode=IN, pull=None):
+        if mode != None:
+            if mode == self.IN:
+                self._mode = self.IN
+                GPIO.setup(self.id, GPIO.IN)
+            elif mode == self.OUT:
+                self._mode = self.OUT
+                GPIO.setup(self.id, GPIO.OUT)
+            else:
+                raise RuntimeError("Invalid mode for pin: %s" % self.id)
+        if pull != None:
+            if self._mode != self.IN:
+                raise RuntimeError("Cannot set pull resistor on output")
+            if pull == self.PULL_UP:
+                GPIO.setup(self.id, GPIO.IN, pull_up_down=GPIO.PUD_UP)
+            elif pull == self.PULL_DOWN:
+                GPIO.setup(self.id, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
+            else:
+                raise RuntimeError("Invalid pull for pin: %s" % self.id)
+
+    def value(self, val=None):
+        if val != None:
+            if val == self.LOW:
+                self._value = val
+                GPIO.output(self.id, val)
+            elif val == self.HIGH:
+                self._value = val
+                GPIO.output(self.id, val)
+            else:
+                raise RuntimeError("Invalid value for pin")
+        else:
+            return GPIO.input(self.id)
+
+    def cleanup(self, channel=None):
+        if channel is None:
+            GPIO.cleanup()
+        elif channel == self:
+            GPIO.cleanup(self.id)
+        else:
+            raise RuntimeError("Invalid pin to cleanup")
+
+D1 = Pin(1)
```

Comment:
this should be the actual name of the underlying Tegra microcontroller, see for example how we do it for allwinner:

https://github.com/adafruit/Adafruit_Blinka/blob/master/src/adafruit_blinka/microcontroller/allwinner_h3/pin.py

`PG13 = Pin(205)` means GPIO205 (which is the libgpiod name) is also known as PG13 in the chip datasheet. basically, the names have to map to the underlying hardware not be named based on the location on a PCB connector

Reply:
The Jetson.GPIO library already does the mapping. The library will not work if the names map to the tegra hardware directly
But I can change the naming of the pins to match the naming on the datasheet

Reply:
you can have a pin id if you like, but given you may change the GPIO library to use libgpiod some day, we should stay true to the hardware

Reply:
Unfortunately, the hardware GPIO cannot be directly accessed due to the nature of the Jetson.GPIO

Jetson.GPIO is designed to resemble RPi.GPIO so they are compatible. It either maps the physical BOARD pin #, BCM pin # as seen on the 40 pin RPi, CVM string name, or TEGRA_SOC string name to the hardware Linux GPIOs

Should the change be made to use libgpiod or update Jetson.GPIO mappings, another PR can be made to reflect this. Until then, there is no way to directly stay true to the hardware

Reply:
OK - please use native chip names in the Tegra **chip** definition, and BCM naming in the board definition, that will make it the most compatible with our existing Pi libraries and examples